x/concentrated-liquidity/lp.go

```diff
@@ -194,23 +194,17 @@ func (k Keeper) WithdrawPosition(ctx sdk.Context, owner sdk.AccAddress, position
 		return sdk.Int{}, sdk.Int{}, err
 	}
 
-	// Retrieve the position in the pool for the provided owner and tick range.
-	positionLiquidity, err := k.GetPositionLiquidity(ctx, positionId)
-	if err != nil {
-		return sdk.Int{}, sdk.Int{}, err
+	// Check if the requested liquidity amount to withdraw is less than or equal to the available liquidity for the position.
+	// If it is greater than the available liquidity, return an error.
+	if requestedLiquidityAmountToWithdraw.GT(position.Liquidity) {
```

Comment:
using position.liquidity directly is a good change! Looked above, and:
> positionHasActiveUnderlyingLockAndUpdate

does not modify the position (only a store for mapping), so I believe this is safe.